src/Mayaqua/DNS.c

```diff
@@ -417,18 +425,31 @@ void DnsResolver(THREAD *t, void *param)
 		for (struct addrinfo *result = results; result != NULL; result = result->ai_next)
 		{
 			IP ip;
-			const struct sockaddr_in6 *in = (struct sockaddr_in6 *)result->ai_addr;
-			InAddrToIP6(&ip, &in->sin6_addr);
-			if (IsIP6(&ip) && ipv6_ok == false)
+			if (hints.ai_family == AF_INET6)
 			{
-				Copy(&resolver->IPv6, &ip, sizeof(resolver->IPv6));
-				resolver->IPv6.ipv6_scope_id = in->sin6_scope_id;
-				ipv6_ok = true;
+				const struct sockaddr_in6 *in = (struct sockaddr_in6 *)result->ai_addr;
+				InAddrToIP6(&ip, &in->sin6_addr);
+				if (IsIP6(&ip) && ipv6_ok == false)
+				{
+					Copy(&resolver->IPv6, &ip, sizeof(resolver->IPv6));
+					resolver->IPv6.ipv6_scope_id = in->sin6_scope_id;
+					ipv6_ok = true;
+				}
+				else if (IsIP4(&ip) && ipv4_ok == false)
+				{
+					Copy(&resolver->IPv4, &ip, sizeof(resolver->IPv4));
+					ipv4_ok = true;
+				}
 			}
-			else if (IsIP4(&ip) && ipv4_ok == false)
+			else
 			{
-				Copy(&resolver->IPv4, &ip, sizeof(resolver->IPv4));
-				ipv4_ok = true;
+				const struct sockaddr_in *in = (struct sockaddr_in *)result->ai_addr;
+				InAddrToIP(&ip, &in->sin_addr);
+				if (IsIP4(&ip) && ipv4_ok == false)
+				{
+					Copy(&resolver->IPv4, &ip, sizeof(resolver->IPv4));
+					ipv4_ok = true;
+				}
```

Comment:
```suggestion
				if (IsIP4(&ip))
				{
					Copy(&resolver->IPv4, &ip, sizeof(resolver->IPv4));
					ipv4_ok = true;
					break;
				}
```

Reply:
Thanks for your review. I have suggested similar changes. What do you think?

Reply:
Yours are much better.

Reply:
They're good.